addons/binding/org.openhab.binding.wso2iots/src/main/java/org/openhab/binding/wso2iots/handler/BridgeHandler.java

```diff
@@ -0,0 +1,219 @@
+            refreshJob = scheduler.scheduleWithFixedDelay(runnable, 0, delay, TimeUnit.MINUTES);
+        }
+    }
+
+    protected BridgeConfiguration getBridgeConfiguration() {
+
+        BridgeConfiguration config = getConfigAs(BridgeConfiguration.class);
+        return config;
+
+    }
+
+    private void checkConnection() throws Exception {
+
+        String errormsg = null;
+        String response1 = null;
+        BridgeConfiguration config = getConfigAs(BridgeConfiguration.class);
+        String hostname = config.hostname;
+
+        String url = "http://" + hostname + ":9763/services/Version.VersionHttpEndpoint";
+
+        URL obj;
+
+        obj = this.validateCertificate(url);
+
+        if (obj == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Certificate validation failed");
+
+        } else {
+
+            HttpURLConnection con = (HttpURLConnection) obj.openConnection();
+
+            con.setRequestMethod("GET");
+            try {
+
+                BufferedReader in = new BufferedReader(new InputStreamReader(con.getInputStream()));
+                String inputLine;
+                StringBuffer response = new StringBuffer();
+
+                while ((inputLine = in.readLine()) != null) {
+                    response.append(inputLine);
+                }
+                in.close();
+
+                response1 = response.toString();
+                logger.debug(response1);
+
+                if (response1.equals("[]")) {
+                    errormsg = "empty response from wso2 iots";
+                    logger.warn(errormsg);
+                }
+            } catch (Exception e) {
+                errormsg = e.getMessage();
+                logger.warn(errormsg);
+            }
+
+            if (errormsg == null) {
+                updateStatus(ThingStatus.ONLINE);
+
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, errormsg);
+
+            }
+        }
+    }
+
+    private URL validateCertificate(String url2) {
+
+        URL url = null;
+
+        TrustManager[] trustAllCerts = new TrustManager[] { new X509TrustManager() {
+            @Override
+            public java.security.cert.X509Certificate[] getAcceptedIssuers() {
+                return null;
+            }
+
+            @Override
+            public void checkClientTrusted(java.security.cert.X509Certificate[] certs, String authType) {
+            }
+
+            @Override
+            public void checkServerTrusted(java.security.cert.X509Certificate[] certs, String authType) {
+            }
+        } };
+
+        // Install the all-trusting trust manager
+        try {
+            SSLContext sc = SSLContext.getInstance("SSL");
+            sc.init(null, trustAllCerts, new java.security.SecureRandom());
+            HttpsURLConnection.setDefaultSSLSocketFactory(sc.getSocketFactory());
```

Comment:
Please don't you might impact / compromise other bindings, if you need you might need to wait for https://github.com/openhab/openhab2-addons/pull/3919 and the related ESH PR to be completed.